shell/src/main/java/org/apache/accumulo/shell/commands/HistoryCommand.java

```diff
@@ -39,11 +39,11 @@
   public int execute(final String fullCommand, final CommandLine cl, final Shell shellState)
       throws IOException {
     if (cl.hasOption(clearHist.getOpt())) {
-      shellState.getReader().getHistory().clear();
+      shellState.getReader().getHistory().purge();
     } else {
-      Iterator<Entry> source = shellState.getReader().getHistory().entries();
+      Iterator<History.Entry> source = shellState.getReader().getHistory().iterator();
       Iterator<String> historyIterator = Iterators.transform(source,
-          input -> String.format("%d: %s", input.index() + 1, input.value()));
+          input -> String.format("%d: %s", input.index() + 1, input.line()));
```

Comment:
It looks like Iterator was used here previously because `.entries()` returned a type of `Iterator`. However, it now looks like `.getHistory()` returns a type of `Iterable`, which gives an opportunity to write this differently, using Java streams or something, and to avoid the use of Guava's `Iterators.transform()` in favor of Java built-ins. It's not important, though. It's fine either way.

Reply:
This is something I can look for additional work after this PR. I did briefly look at trying to implement it and noticed a decent amount of code to change at where we interpret the results and print to shell.